crates/core/transaction/src/view.rs

```diff
@@ -94,6 +99,173 @@ impl TransactionView {
+                        claim_tx: _,
+                        asset_1_metadata: _,
+                        asset_2_metadata: _,
+                        batch_swap_output_data: _,
+                    } => {
+                        let address = AddressView::Opaque {
+                            address: output_1.clone().expect("sender address").address(),
+                        };
+
+                        let value_fee = Value {
+                            amount: swap_plaintext.claim_fee.amount(),
+                            asset_id: swap_plaintext.claim_fee.asset_id(),
+                        };
+                        let value_1 = Value {
+                            amount: swap_plaintext.delta_1_i,
+                            asset_id: swap_plaintext.trading_pair.asset_1(),
+                        };
+                        let value_2 = Value {
+                            amount: swap_plaintext.delta_2_i,
+                            asset_id: swap_plaintext.trading_pair.asset_2(),
+                        };
+
+                        // Required imbalance (-)
+                        let mut balance = Balance::default();
+                        balance -= value_1;
+                        balance -= value_2;
+                        balance -= value_fee;
+
+                        effects.push(TransactionEffect { address, balance });
+                    }
+                    SwapView::Opaque {
+                        swap: _,
+                        batch_swap_output_data: _,
+                        output_1: _,
+                        output_2: _,
+                        asset_1_metadata: _,
+                        asset_2_metadata: _,
+                    } => continue,
+                },
+                ActionView::SwapClaim(swap_claim_view) => match swap_claim_view {
+                    SwapClaimView::Visible {
+                        swap_claim,
+                        output_1,
+                        output_2: _,
+                        swap_tx: _,
+                    } => {
+                        let address = AddressView::Opaque {
+                            address: output_1.address(),
+                        };
+
+                        let value_fee = Value {
+                            amount: swap_claim.body.fee.amount(),
+                            asset_id: swap_claim.body.fee.asset_id(),
+                        };
+
+                        // Provided imbalance (+)
+                        let mut balance = Balance::default();
+                        balance += value_fee;
+
+                        effects.push(TransactionEffect { address, balance });
+                    }
+                    SwapClaimView::Opaque { swap_claim: _ } => continue,
+                },
+                _ => {} // Fill in other action views as neccessary
+            }
+        }
+
+        let mut summary = TransactionSummary { effects };
+
+        Self::accumulate_summary(&mut summary)
```

Comment:
```suggestion
        let summary = TransactionSummary { effects };

        Self::accumulate_effects(summary)
```

(If you implement the suggestion above, this will need to change too)